homeassistant/components/imap/quality_scale.yaml

```diff
@@ -0,0 +1,96 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup:
+    status: exempt
+    comment: >
+      Per IMAP service instance there is one numeric sensor entity to reflect
+      the actual number of emails for a service. There is no event registration.
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  config-entry-unloading: done
+  log-when-unavailable:
+    status: done
+    comment: |
+      Logs for unavailability are on debug level to avoid flooding the logs.
+  entity-unavailable:
+    status: done
+    comment: >
+      An entity is available as long as the service is loaded.
+      An `unknown` value is set if the mail service is temporary unavailable.
+  action-exceptions: done
+  reauthentication-flow: done
+  parallel-updates: done
+  test-coverage: done
+  integration-owner: done
+  docs-installation-parameters: done
+  docs-configuration-parameters: done
+
+  # Gold
+  entity-translations: done
+  entity-device-class: done
+  devices: done
+  entity-category: done
+  entity-disabled-by-default:
+    status: done
+    comment: The only entity supplied returns the primary value for the service.
+  discovery:
+    status: exempt
+    comment: |
+      Discovery for IMAP services is not desirerable.
+  stale-devices:
+    status: exempt
+    comment: >
+      The device class is a service. When removed, entities are removed as well.
+  diagnostics: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow:
+    status: exempt
```

Comment:
I don't think it is exemptable.

Setting up this config entry, provides all kinds of data a user might want to be able to change. For example, moving ports or server addresses or adjustments to SSL properties.

Reply:
Well all items that should be configurable are reconfugrable through the option flow. So may be this one should be set to `done`?

Reply:
> Well all items that should be configurable are reconfugrable through the option flow. 

That sounds incorrect. The options flows provides options, while the configuration flow is for connection and authentication details.

If this is mixed up, it would mean the `config-flow` rule also isn't following the guidelines (and thus should be marked `todo` as well).

Reply:
You are completely right. Thanks for you patience. It seems this part is missing for IMAP.